src/storage/txn/task.rs

```diff
@@ -56,13 +61,27 @@ impl Task {
         self.extra_op = extra_op
     }
 
+    pub(super) fn alloc_memory_quota(
+        &mut self,
+        memory_quota: Arc<MemoryQuota>,
+    ) -> Result<(), MemoryQuotaExceeded> {
+        let bytes = self.cmd.heap_size();
+        memory_quota.alloc(bytes)?;
+        self.memory_quota = Some((memory_quota, bytes));
+        Ok(())
+    }
+
     pub(super) fn process_write<S: Snapshot, L: LockManager>(
         mut self,
         snapshot: S,
         context: WriteContext<'_, L>,
     ) -> super::Result<WriteResult> {
         let cmd = self.cmd.take().unwrap();
-        cmd.process_write(snapshot, context)
+        let res = cmd.process_write(snapshot, context);
+        if let Some((memory_quota, bytes)) = self.memory_quota.take() {
+            memory_quota.free(bytes);
+        }
```

Comment:
Can this be a separated function?  Or maybe we can also make the `memory_quota` field an RAII guard so that we won't forget free it accidentally in the future.

Reply:
We have implemented the RAII style `free` in Task's `Drop`.

Reply:
I mean, I thought it might be easier to use if the RAII is done on the `memory_quota` field instead of `Task`. So that when we need to free it manually, we can just take it out without any additional code. It can even prevent leak if we forget to free the previous one when allocation is done more than once (maybe by mistake).
This is not important though, as this kind of mistake doesnt look so likely to happen...